### PR TITLE
feat: implement and register GraviteeProductInitializer for APIM Gateway

### PR DESCRIPTION
**Pull request by benoitgravitee**
## Issue

https://gravitee.atlassian.net/browse/ARCHI-353

## Description

Implement product initializer for APIM Gateway (used in `gravitee-cloud-initializer`)

## Additional context

https://github.com/gravitee-io/gravitee-node/pull/356
https://github.com/gravitee-io/gravitee-cloud-initializer/pull/22

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jxqlrslzod.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/pom.xml [hunk 1]
```diff
@@ -67,11 +67,6 @@
 			<version>${project.version}</version>
 		</dependency>
 
-		<dependency>
```
Comment (benoitgravitee) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/pom.xml:70]: Duplication spotted by IDE hence the cleanup

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/pom.xml [hunk 2]
```diff
@@ -142,10 +142,6 @@
 			<version>${project.version}</version>
 		</dependency>
 
-		<dependency>
```
Comment (benoitgravitee) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/pom.xml:145]: Duplication spotted by IDE hence the cleanup

File: gravitee-apim-rest-api/gravitee-apim-rest-api-repository/pom.xml [hunk 3]
```diff
@@ -45,11 +45,6 @@
 			<artifactId>gravitee-node-api</artifactId>
 		</dependency>
 
-		<dependency>
```
Comment (benoitgravitee) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-repository/pom.xml:48]: Duplication spotted by IDE hence the cleanup

File: gravitee-apim-rest-api/pom.xml [hunk 4]
```diff
@@ -75,26 +75,6 @@
             <artifactId>commons-lang3</artifactId>
         </dependency>
 
-        <dependency>
```
Comment (benoitgravitee) [inline gravitee-apim-rest-api/pom.xml:78]: Duplications spotted by IDE hence the cleanup

File: pom.xml [hunk 5]
```diff
@@ -67,7 +67,7 @@
         <gravitee-fetcher-api.version>2.0.0</gravitee-fetcher-api.version>
         <gravitee-gateway-api.version>3.5.1</gravitee-gateway-api.version>
         <gravitee-integration-api.version>4.1.0</gravitee-integration-api.version>
-        <gravitee-node.version>6.0.4</gravitee-node.version>
+        <gravitee-node.version>6.3.0-archi-352-identify-workload-SNAPSHOT</gravitee-node.version>
```
Comment (benoitgravitee) [inline pom.xml]: update before merge

File: pom.xml [hunk 6]
```diff
@@ -80,7 +80,7 @@
         <gravitee-service-discovery-api.version>1.1.1</gravitee-service-discovery-api.version>
         <gravitee-kubernetes.version>3.3.1</gravitee-kubernetes.version>
         <gravitee-json-validation.version>2.0.1</gravitee-json-validation.version>
-        <gravitee-cloud-initializer.version>1.0.0</gravitee-cloud-initializer.version>
+        <gravitee-cloud-initializer.version>1.1.0-archi-352-support-fed-agent-SNAPSHOT</gravitee-cloud-initializer.version>
```
Comment (benoitgravitee) [inline pom.xml]: update before merge

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/java/io/gravitee/gateway/standalone/APIMGatewayProductInitializer.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * @author Benoit BORDIGONI (benoit.bordigoni at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class APIMGatewayProductInitializer implements GraviteeProductInitializer {
+
+    @Override
+    public String productName() {
+        return "apim-gateway";
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/java/io/gravitee/gateway/standalone/APIMGatewayProductInitializer.java]: As discussed 'gio-apim-gateway'

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/resources/META-INF/spring.factories [hunk 8]
```diff
@@ -1,2 +1,4 @@
 io.gravitee.node.api.healthcheck.Probe=\
-    io.gravitee.gateway.standalone.healthcheck.HttpServerProbe
\ No newline at end of file
+    io.gravitee.gateway.standalone.healthcheck.HttpServerProbe
+io.gravitee.node.container.ContainerInitializer=\
+    io.gravitee.gateway.standalone.APIMGatewayProductInitializer
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/resources/META-INF/spring.factories:4]: Is it necessary?

**Review by jhaeyaert — CHANGES_REQUESTED** [review]
Should it be implemented for management as well?

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/resources/META-INF/spring.factories (hunk 8 above)
Comment (benoitgravitee) [reply gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/resources/META-INF/spring.factories:4]: That is how it is Initializing bean work in Node I just kept the way it is done.

Comment (benoitgravitee) [reply gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/resources/META-INF/spring.factories:4]: It is mandatory... simple bean don't work.

**Review by jgiovaresco — APPROVED** [review]

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**